nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -705,6 +705,35 @@ export async function findUsableCommitmentsMutex(
   );
 }
 
+/**
+ *
+ * @function saveCommitments save a list of commitments in the database
+ * @param {[]} listOfCommitments a list of commitments to be saved in the database
+ * @returns
+ */
+export async function saveCommitments(listOfCommitments) {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+  let exist = false;
+  for (const commitment of listOfCommitments) {
+    if (
+      // eslint-disable-next-line no-await-in-loop
+      (await db.collection(COMMITMENTS_COLLECTION).find({ _id: commitment._id }).toArray()).length >
+      0
+    ) {
+      exist = true;
+      break;
+    }
+  }
+
+  if (exist) {
```

Comment:
Why do we raise error if duplicated commitment? Can’t you just not copy that one?

Reply:
I'm thinking to run a map in the list of commitments and just get those that doesn't exists in the database yet.
This way we just save new commitments.
Because the insertMany operation save all until the find the first that already exists.